web/src/main/java/co/mobileaction/example/web/repository/IPostRepository.java

```diff
@@ -13,4 +17,9 @@
 public interface IPostRepository extends JpaRepository<Post, Long>
 {
     List<Post> findAllByUserId(Long userId);
+
+    @Modifying
+    @Transactional
+    @Query("DELETE FROM Post p WHERE p.userId = :userId")
+    void deleteAllByUserId(@Param("userId") Long userId);

```

Comment:
for the large number of posts, we could use native query (inside `Query` annotation, we can set `nativeQuery = true`) because hibernate first gets the entities, than deletes it row by row (so could be inefficient). But, this is ok for the demo project, just wanted you to know about it.